cover_agent/CoverageProcessor.py

```diff
@@ -222,3 +252,11 @@ def extract_package_and_class_java(self):
             raise
 
         return package_name, class_name
+
+    def get_file_extension(self, filename: str) -> str | None:
+        """Get the file extension from a given filename."""
+        match = re.search(r'\.(\w+)$', filename)
+        if match:
+            return match.group(1)
+        else:
+            return None
```

Comment:
I'm pretty sure you can just use an `os` (or similar) library to extract out a path name. No need for this function.